ftdc/ftdc.go

```diff
@@ -190,7 +225,15 @@ func (ftdc *FTDC) writeDatum(datum datum) error {
 	// The input `datum` being processed is for a different schema than we were previously using.
 	if datum.generationID != ftdc.outputGenerationID {
 		// Compute the new schema and write that to disk.
-		ftdc.currSchema = getSchema(datum.Data)
+		newSchema, schemaErr := getSchema(datum.Data)
+		if schemaErr != nil {
+			ftdc.logger.Warnw("Could not generate schema for statser",
+				"statser", schemaErr.statserName, "err", schemaErr.err)
+			ftdc.conditionalRemoveStatser(schemaErr.statserName, datum.generationID)
```

Comment:
The business logic to get the list of registered statsers into a "good" state of only well-behaved statsers.